lib/src/services/common-service.dart

```diff
@@ -0,0 +1,131 @@
+part of 'tmdb-service.dart';
+
+abstract class _CommonService<T> {
```

Comment:
This class has lot of switch cases on different types, which is a violation of the Open Closed Principle.
I would suggest adding the model factories as parameters to the methods, and passing them from MovieService and TvService respectively.
Moreover, the type parameter on the class is used only in some of the methods while other have their own constraint, to leave maximum flexibility I suggest to remove the type parameter from the class level and only have type parameter on method level.